whenchanged/whenchanged.py

```diff
@@ -121,33 +122,30 @@ def is_interested(self, path):
 
         return False
 
-    def on_change(self, path):
+    def on_change(self, path, event = None):
+        if event is not None and not event.is_directory:
+            self.last_event_type = event.event_type
+            self.set_envvar('event', 'file_' + event.event_type)
         if self.is_interested(path):
             self.run_command(path)
 
     def on_created(self, event):
-        if self.observer.__class__.__name__ == 'InotifyObserver':
-            # inotify also generates modified events for created files
-            return
-
         if not event.is_directory:
-            self.set_envvar('event', 'file_created')
-            self.on_change(event.src_path)
+            self.on_change(event.src_path, event)
 
     def on_modified(self, event):
-        if not event.is_directory:
-            self.set_envvar('event', 'file_modified')
-            self.on_change(event.src_path)
+        if not event.is_directory and self.last_event_type != 'created':
+            self.on_change(event.src_path, event)
+        if not event.is_directory and self.last_event_type == 'created':
+            self.last_event_type = event.event_type
 
     def on_moved(self, event):
         if not event.is_directory:
-            self.set_envvar('event', 'file_moved')
-            self.on_change(event.dest_path)
+            self.on_change(event.src_path, event)
 
```

Comment:
Why dest_path -> src_path?

Reply:
Sorry, I had a conflict while unstashing after rebased and missed this issue.
Done